ironfish/src/network/peers/peerManager.ts

```diff
@@ -783,55 +796,54 @@ export class PeerManager {
   }
 
   start(): void {
-    this.distributePeerListHandle = setInterval(() => this.distributePeerList(), 5000)
+    this.requestPeerListHandle = setInterval(() => this.requestPeerList(), 60000)
     this.disposePeersHandle = setInterval(() => this.disposePeers(), 2000)
+    this.savePeerAddressesHandle = setInterval(
+      () => void this.addressManager.save(this.peers),
+      60000,
+    )
   }
 
   /**
    * Call when shutting down the PeerManager to clean up
    * outstanding connections.
    */
-  stop(): void {
-    this.distributePeerListHandle && clearInterval(this.distributePeerListHandle)
+  async stop(): Promise<void> {
+    this.requestPeerListHandle && clearInterval(this.requestPeerListHandle)
     this.disposePeersHandle && clearInterval(this.disposePeersHandle)
+    this.savePeerAddressesHandle && clearInterval(this.savePeerAddressesHandle)
+    await this.addressManager.save(this.peers)
     for (const peer of this.peers) {
       this.disconnect(peer, DisconnectingReason.ShuttingDown, 0)
     }
   }
 
-  private distributePeerList() {
-    const connectedPeers = []
-
-    for (const p of this.identifiedPeers.values()) {
-      if (p.state.type !== 'CONNECTED') {
-        continue
-      }
-
-      connectedPeers.push({
-        identity: p.state.identity,
-        name: p.name || undefined,
-        address: p.address,
-        port: p.port,
-      })
-    }
-
-    const peerList: PeerList = {
-      type: InternalMessageType.peerList,
-      payload: { connectedPeers },
-    }
-
+  private requestPeerList() {
     const peerListRequest: PeerListRequest = {
       type: InternalMessageType.peerListRequest,
     }
 
     for (const peer of this.getConnectedPeers()) {
-      if (peer.version !== null && peer.version >= MIN_VERSION_FOR_PEER_LIST_REQUESTS) {
-        peer.send(peerListRequest)
-        continue
-      }
+      peer.send(peerListRequest)
+    }
+  }
 
-      peer.send(peerList)
+  /**
+   * Gets a random disconnected peer address and returns a peer created from
+   * said address
+   */
+  createRandomDisconnectedPeer(): Peer | null {
+    const peerAddress = this.addressManager.getRandomDisconnectedPeerAddress(
+      Array.from(this.identifiedPeers.keys()),
```

Comment:
Is this value supposed to be only connected/connecting peers? I'm pretty sure identifiedPeers can include disconnected peers as well, so this should be filtered by the peer state.

Reply:
It is, so I'll do a filter. Thanks!